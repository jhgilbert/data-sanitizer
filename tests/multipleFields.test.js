const { sanitizeFields } = require("..")

test('sanitizes multiple fields', () => {
  let data = [
    { 
      id: 1, 
      name: 'Jen Gilbert', 
      pets: [
        { id: 3, name: 'Frank' }
      ]
    },
    { 
      id: 2, 
      name: 'Derek Reeve', 
      pets: [
        { id: 4, name: 'Evey' },
        { id: 5, name: 'Alfie' }
      ]
    }
  ]

  const sanitationResult = sanitizeFields({ data, fieldNames: ['pets', 'id'] })
  expect(sanitationResult).toMatchSnapshot()
})