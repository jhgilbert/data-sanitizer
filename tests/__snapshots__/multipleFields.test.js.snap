// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sanitizes multiple fields 1`] = `
Object {
  "data": Array [
    Object {
      "id": "[SANITIZED]",
      "name": "Jen Gilbert",
      "pets": "[SANITIZED]",
    },
    Object {
      "id": "[SANITIZED]",
      "name": "Derek Reeve",
      "pets": "[SANITIZED]",
    },
  ],
  "removedValues": Object {
    "id": Array [
      1,
      2,
    ],
    "pets": Array [
      Array [
        Object {
          "id": 3,
          "name": "Frank",
        },
      ],
      Array [
        Object {
          "id": 4,
          "name": "Evey",
        },
        Object {
          "id": 5,
          "name": "Alfie",
        },
      ],
    ],
  },
}
`;
